packaging/release.py

```diff
@@ -0,0 +1,1410 @@
+
+    if dirty:
+        with suppress_when(allow_dirty):
+            raise ApplicationError(f"There are {len(dirty)} files which are untracked and/or have changes, which will be omitted from the build.")
+
+    sdist_file = get_sdist_path(version)
+    wheel_file = get_wheel_path(version)
+
+    with tempfile.TemporaryDirectory(dir=DIST_DIR, prefix=f"build-{version}-", suffix=".tmp") as temp_dir_name:
+        temp_dir = pathlib.Path(temp_dir_name)
+        dist_dir = temp_dir / "dist"
+
+        git("worktree", "add", "-d", temp_dir)
+
+        try:
+            run("python", "-m", "build", "--config-setting=--build-manpages", env=env, cwd=temp_dir)
+
+            get_sdist_path(version, dist_dir).rename(sdist_file)
+            get_wheel_path(version, dist_dir).rename(wheel_file)
+        finally:
+            git("worktree", "remove", temp_dir)
+
+
+@command
+def test() -> None:
+    """Test the sdist and wheel."""
+    command.run(
+        test_sdist,
+        test_wheel,
+    )
+
+
+@command
+def test_sdist() -> None:
+    """Test the sdist."""
+    version = get_ansible_version(mode=VersionMode.ALLOW_DEV_POST)
+    sdist_file = get_sdist_path(version)
+
+    with tempfile.TemporaryDirectory() as temp_dir_name:
+        temp_dir = pathlib.Path(temp_dir_name)
+
+        with contextlib.ExitStack() as stack:
+            try:
+                sdist = stack.enter_context(tarfile.open(sdist_file))
+            except FileNotFoundError:
+                raise ApplicationError(f"Missing sdist: {sdist_file.relative_to(CHECKOUT_DIR)}") from None
+
+            sdist.extractall(temp_dir)
+
+        man1_dir = temp_dir / sdist_file.with_suffix("").with_suffix("").name / "docs" / "man" / "man1"
+        man1_pages = sorted(man1_dir.glob("*.1"))
+
+        if not man1_pages:
+            raise ApplicationError(f"No man pages found in the sdist at: {man1_dir.relative_to(temp_dir)}")
+
+        pyc_glob = "*.pyc*"
+        pyc_files = sorted(path.relative_to(temp_dir) for path in temp_dir.rglob(pyc_glob))
+
+        if pyc_files:
+            raise ApplicationError(f"Found {len(pyc_files)} '{pyc_glob}' file(s): {', '.join(map(str, pyc_files))}")
+
+        display.show(f"Found man1 pages: {', '.join([path.name for path in man1_pages])}")
+
+    test_built_artifact(sdist_file)
+
+
+@command
+def test_wheel() -> None:
+    """Test the wheel."""
+    version = get_ansible_version(mode=VersionMode.ALLOW_DEV_POST)
+    wheel_file = get_wheel_path(version)
+
+    with tempfile.TemporaryDirectory() as temp_dir_name:
+        temp_dir = pathlib.Path(temp_dir_name)
+
+        with contextlib.ExitStack() as stack:
+            try:
+                wheel = stack.enter_context(zipfile.ZipFile(wheel_file))
+            except FileNotFoundError:
+                raise ApplicationError(f"Missing wheel for version {version}: {wheel_file}") from None
+
+            wheel.extractall(temp_dir)
```

Comment:
Does this just smoke-test that unpacking won't fail? Why not use the same mechanism pip would use, then?
```suggestion
        run('wheel', 'unpack', str(wheel_file), cwd=temp_dir)
```

Reply:
There's already a separate check using `pip install` to make sure the wheel (and sdist) can be installed.

This check is done first using `extractall`, where the error handling is easier than using a subprocess. Having the contents in a temporary directory will make adding additional checks in the future easier (something already done for the sdist test).